lib/sentry/client_error.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Sentry.ClientError do
+  @moduledoc """
+  An exception struct that represents an error returned by Sentry.
+
+  This struct is designed to manage and handle errors originating from operations
+  in the client application. The `:reason` field signifies the cause of the error
+  as an atom or tuple.
+
+  To raise instances of this exception, you can use `Kernel.raise/1`. When crafting
+  formatted error messages for purposes such as logging or presentation, consider
+  leveraging `Exception.message/1`.
+  """
+
+  @moduledoc since: "10.7.0"
+
+  @doc """
+  The exception struct for a Sentry error.
+  """
+  defexception [:reason]
+
+  @typedoc """
+  The type for a Sentry error exception.
+  """
+  @type t() :: %__MODULE__{reason: term}
+
+  @doc false
+  @spec new(term) :: t
+  def new(reason) do
+    %__MODULE__{reason: reason}
+  end
+
+  @impl true
+  def message(%__MODULE__{reason: reason}) do
+    "Request failure reason: #{format(reason)}"
```

Comment:
Nit, but usually in Elixir we start exception messages with lowercase letters.

It would be

```suggestion
    "request failure reason: #{format(reason)}"
```

here. Can you fix that up in this file?